BrainPortal/app/views/data_providers/show.html.erb

```diff
@@ -42,9 +42,14 @@
                   :generate         => "ok"
                 ), :class => "button" %>
 
-  <% if check_role(:admin_user) || @provider.user_id == current_user.id %>
-    <%= link_to 'Delete', data_provider_path(@provider), { :data => { :confirm => "Are you sure you want to delete '#{@provider.name}' ?" }, :method => :delete, :class => "button" } %>
-  <% end %>
+
+    <% if @provider.type == "UserkeyFlatDirSshDataProvider" %>
```

Comment:
Don't do equality checks on object types, use `.is_a?` instead.

Reply:
Also, maybe better is to verify the capability of the data provider. If it has the check() method.

Reply:
good idea, but now we should ow not use check for any other purposes and maintain aryty